### PR TITLE
For i03-365 - look for cutoff in meta file

### PR DESCRIPTION
**Pull request by graeme-winter**
At the moment this is not linkable from the proper h5 file structure however
is included in the meta file as the config dataset, a text file which
comes from the detector - look this up as a workaround

If *anything* goes wrong, ignore by design

### CODE REVIEW

File: format/FormatNexusEigerDLS16M.py [hunk 1]
```diff
@@ -51,6 +61,28 @@ def __init__(self, image_file, **kwargs):
         super(FormatNexusEigerDLS16M, self).__init__(image_file, **kwargs)
         self._dynamic_shadowing = self.has_dynamic_shadowing(**kwargs)
 
+    def get_detector(self, index=None):
+        if "_master.h5" not in self._image_file:
```
Comment (Anthchirp) [inline format/FormatNexusEigerDLS16M.py]: should this be 
```suggestion
        if not self._image_file.endswith("_master.h5"):
```

File: format/FormatNexusEigerDLS16M.py [hunk 2]
```diff
@@ -51,6 +61,28 @@ def __init__(self, image_file, **kwargs):
...
+
+        # workaround for https://jira.diamond.ac.uk/browse/I03-365
+        # read the count limit from the meta file - if anything goes
+        # wrong, do nothing
+
+        detector = self._detector()
+
+        try:
+            meta = self._image_file.replace("_master.h5", "_meta.h5")
+            limit = get_count_limit_from_meta(meta)
```
Comment (Anthchirp) [inline format/FormatNexusEigerDLS16M.py]: should we check for nonsensical values? `if limit > 0: for ...`?

Comment (graeme-winter) [reply format/FormatNexusEigerDLS16M.py]: Yes, probably

Comment (graeme-winter) [reply format/FormatNexusEigerDLS16M.py]: Added assert that limit > 0

**Pull request merged**